libgrease/src/channel_metadata.rs

```diff
@@ -20,18 +20,53 @@ pub struct ChannelMetadata {
     /// The channel ID
     channel_id: ChannelId,
     /// The KES identifier.
-    kes_public_key: String,
+    kes_public_key: GenericPoint,
+    /// The external identifier
+    public_key_self: GenericPoint,
+    /// The key to the external identifier.
+    private_key_self: GenericScalar,
+    /// Random nonce
+    nonce_self: GenericScalar,
+    ///
+    public_key_peer: GenericPoint,
+    ///
+    nonce_peer: GenericScalar,
```

Comment:
It's not clear to me why these belong in metadata. Are they referred to throughout the lifecycle of the Channel?
Perhaps you meant to put them in ChannelProposal?

Reply:
We need the peer's public nonce to exist before the proving begins to ensure no reuse. It's an interactive step that's hard to wedge into the existing design. My quick fix was to throw that into the original channel definition since it won't hurt anything to have it declared at the beginning. Ideally it would be part of the first round of updates, but that requires reworking a bunch of things to support state mutation.